### PR TITLE
Test Info report in connectivity test

### PR DESCRIPTION
**Pull request by philipktlin**
Add new test info report and report network frequencies from network controller module.

### CODE REVIEW

**Comment by dylanbronson** [comment]
What is TestInfo for? Will it be used for more than just NetworkController?

File: test/connectivity/modules/NetworkController/Program.cs [hunk 1]
```diff
@@ -23,6 +26,8 @@ static async Task Main()
             {
                 var networkInterfaceName = DockerHelper.GetDockerInterfaceName();
 
+                await ReportTestInfoAsync();
```
Comment (dylanbronson) [inline test/connectivity/modules/NetworkController/Program.cs]: Move to after testDelay

File: test/connectivity/modules/NetworkController/Program.cs [hunk 2]
```diff
@@ -67,23 +70,51 @@ static async Task Main()
             handler.ForEach(h => GC.KeepAlive(h));
         }
 
+        static async Task ReportTestInfoAsync()
+        {
+            var testInfoResults = new string[]
```
Comment (ancaantochi) [inline test/connectivity/modules/NetworkController/Program.cs]: why isn't it reporting just once with all the settings?

File: test/modules/TestResultCoordinator/Reports/TestInfoReportMetadata.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports
+{
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+
+    class TestInfoReportMetadata : ITestReportMetadata
+    {
+        public TestInfoReportMetadata()
```
Comment (ancaantochi) [inline test/modules/TestResultCoordinator/Reports/TestInfoReportMetadata.cs]: is this constructor needed?

**Pull request merged**